src/main/resources/hudson/tasks/junit/History/history.js

```diff
@@ -323,12 +323,16 @@ function onBuildIntervalChange(selectObj) {
                     top: '20%',
                 },
                 xAxis: {
-                    type: 'category',
-                    boundaryGap: false,
+                    type: 'value',
                     axisLabel: {
-                        color: textColor
+                        color: textColor,
+                        formatter: function(value) {
+                            return Math.round(value * model.distribution.xAxis.roundingFactor) / model.distribution.xAxis.roundingFactor;
+                        }
                     },
-                    data: model.distribution.domainAxisLabels,
+                    min: model.distribution.xAxis.min,
+                    max: model.distribution.xAxis.max,
+                    minInterval: model.distribution.xAxis.interval,
```

Comment:
What I really wanted was to be able to tell `echarts` to pick an interval that is a multiple of `model.distribution.xAxis.interval`, and then let it automatically choose which multiple based on screen and label size, but I didn't see any way to do it.

If you set `interval: model.distribution.xAxis.interval`, then you get one grid line per bucket, which is nice, but the labels are unreadably close to each other and I couldn't find a way to fix that dynamically. You can set `interval: model.distribution.xAxis.interval * 2` and things look fine on "normalish" laptop screen size, but again it doesn't scale at all if you shrink your screen size.

I also wanted to have dynamic `minorTick` support, e.g. if we let `echarts` pick the interval, could we at least set up the minor ticks to align with the buckets? Unfortunately I don't think so, you can only control how many minor ticks you want between each major tick, e.g. if you specify `interval: model.distribution.xAxis.interval * 3`, you could specify `minorTick { show: true, splitNumber: 3 }`, but again that doesn't adjust based on the data or screen size at all.